File: rust/src/models/usage_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::metric_usage::MetricUsage;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct UsageResponse {
    pub period_end: String,

    pub period_start: String,

    pub usage: Vec<MetricUsage>,
}

impl UsageResponse {
    pub fn new(period_end: String, period_start: String, usage: Vec<MetricUsage>) -> Self {
        Self {
            period_end,
            period_start,
            usage,
        }
    }
}

File: typescript/src/models/usageResponse.ts
// this file is @generated
import { type MetricUsage, MetricUsageSerializer } from "./metricUsage";

export interface UsageResponse {
  periodEnd: string;

  periodStart: string;

  usage: MetricUsage[];
}

export const UsageResponseSerializer = {
  _fromJsonObject(object: any): UsageResponse {
    return {
      periodEnd: object["period_end"],
      periodStart: object["period_start"],
      usage: object["usage"].map((item: any) =>
        MetricUsageSerializer._fromJsonObject(item)
      ),
    };
  },

  _toJsonObject(self: UsageResponse): any {
    return {
      period_end: self.periodEnd,
      period_start: self.periodStart,
      usage: self.usage.map((item: any) => MetricUsageSerializer._toJsonObject(item)),
    };
  },
};
